File: switch_if_alternative/if_switch_must_go.cpp
#include <iostream>
#include <map>


std::string one(){  return "I am one "; }
std::string two() { return "Now this was option 2"; }
std::string three() { return "You chose option 3"; }

/*
* example of using function pointers in arrays and maps (hashes) to clean up code and avoid
* large switch or if blocks.
*/

int main(){
  std::string choice="";

  //declare a map (hash) of 3 function pointers that take int as argument. C++ as std::function which would be a cleaner more c++ approach for prototyping def.
  std::map<std::string, std::string (*)()> functions = { {"one", &one}, {"1", &one}, {"two", &two}, {"2", &two}, {"three", &three}, {"3", &three} };

  std::cout << "enter: one, two or three" << std::endl;
  std::cin >> choice;

  std::cout << ( (functions.find(choice)!=functions.end()) ? functions[choice]() : "Whoops wrong input motherfucker" );  
}

File: switch_if_alternative/if_switch_must_go.js
const readline = require('readline-sync');

const one = () => "This is option 1";
const two = () => "Now this was option 2";
const three = () => "You chose option 3";

const choice =  readline.question('Enter: one, two or three: \n');

switch(choice){
    case "1":
        console.log(one());
        break;
    case "one":
        console.log(one());    
        break;
    case "2":
        console.log(two());
        break;
    case "two":
        console.log(two());
        break;
    case "3":
        console.log(three());
        break;
    case "three":
        console.log(three());
        break;
    default:
        console.log("whoops motherfucker!");
}

//this is shorter and cleaner
functions = { "1": one, "one": one, "2": two, "two": two, "3": three, "three": three}
console.log( (choice in functions) ? functions[choice]() : "whoops motherfucker!");